server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStatusApisIT.java

```diff
@@ -189,4 +193,102 @@ public void testGetSnapshotsWithoutIndices() {
         assertThat(found.snapshotId(), is(snapshotInfo.snapshotId()));
         assertThat(found.state(), is(SnapshotState.SUCCESS));
     }
+
+    public void testCorrectCountsForDoneShards() throws Exception {
```

Comment:
can you add some javadocs to this method to describe what exact situation is being tested here? There are a large number of steps in the test, and it might be easier to have a high-level  description of what's  going on  here.